pkg/agent/route/route_linux.go

```diff
@@ -417,6 +421,41 @@ func (c *Client) syncRoute() error {
 	return nil
 }
 
+func (c *Client) syncNeighbor() error {
+	neighborList, err := c.netlink.NeighList(c.nodeConfig.GatewayConfig.LinkIndex, netlink.FAMILY_ALL)
+	if err != nil {
+		return err
+	}
+	neighborKeys := sets.New[string]()
+	for i := range neighborList {
+		n := neighborList[i]
+		if n.State != netlink.NUD_PERMANENT {
+			continue
+		}
+		neighborKeys.Insert(n.IP.String())
+	}
+	restoreNeighbor := func(neighbor *netlink.Neigh) bool {
+		if neighborKeys.Has(neighbor.IP.String()) {
```

Comment:
in theory, shouldn't we check the `HardwareAddr` as well? For routes, we include other attributes in the "key", not just the route prefix.
If the `HardwareAddr` was modified on the system by mistake, we probably want to override that here.

Reply:
Done.